tests/test_utils.py

```diff
@@ -201,3 +202,27 @@ def test_obfuscate_sensitive_headers(headers, output):
     bytes_output = [(k.encode(), v.encode()) for k, v in output]
     assert list(obfuscate_sensitive_headers(headers)) == output
     assert list(obfuscate_sensitive_headers(bytes_headers)) == bytes_output
+
+
+@pytest.mark.parametrize(
+    ["url", "no_proxy", "expected"],
+    [
+        ("http://127.0.0.1", {"NO_PROXY": ""}, False),
+        ("http://127.0.0.1", {"NO_PROXY": "127.0.0.1"}, True),
+        ("http://127.0.0.1", {"NO_PROXY": "https://127.0.0.1"}, False),
+        ("http://127.0.0.1", {"NO_PROXY": "1.1.1.1"}, False),
+        ("http://courses.mit.edu", {"NO_PROXY": "mit.edu"}, True),
+        ("https://mit.edu.info", {"NO_PROXY": "mit.edu"}, False),
+        ("https://mit.edu.info", {"NO_PROXY": "mit.edu,edu.info"}, True),
```

Comment:
Can we add a test case just above for `("https://mit.edu.info", {"NO_PROXY": "edu.info"}, True)`,to make extra clear that the `edu.info` being contained in `mit.edu.info` is what causes us to decide not to proxy this URL?

Reply:
sure, though test at line `214` with `http://courses.mit.edu` and `mit.edu` meets that branch
i could also add comments for what each test does

Reply:
Oh absolutely, I didn’t see the test on L214.